core/src/main/cljc/jarl/gencaps.cljc

```diff
@@ -0,0 +1,20 @@
+(ns jarl.gencaps
+  (:require [jarl.builtins.registry :as registry]
+            [clojure.core :as core]
+            [clojure.set :as set]))
+
+(def supported (set (keys registry/builtins)))
+
+;; Import capabilities.json as a Clojure data structure
+(def capabilities (-> "capabilities.json"
+                      core/slurp
+                      (core/str)
+                      core/read-string))
```

Comment:
The error you're seeing is from this line — the `read-string` function is for Clojure data, not JSON. You can require `jarl.encoding.json :as json` and then use `json/read-str` here instead 🙂